agents_infra/agents_infra/utils/communication.py

```diff
@@ -0,0 +1,32 @@
+from logtools import maybe_log_message
+from token_manager import TokenManager
+
+
+class AgentCommunicator(object):
+    def __init__(self, server_name, post_data_fn, logger):
+        self.server_name = server_name
+        self.post_data = post_data_fn
+        self.logger = logger
+        self.token_manager = TokenManager(server_name, post_data_fn, logger)
+
+    def post_data_with_auth(self, url, payload, **kwargs):
```

Comment:
Add a docstring describing what this method does, its parameters, and what it returns or raises.